vimdoc/module.py

```diff
@@ -149,19 +149,60 @@ def Close(self):
     for backmatter in self.backmatters:
       if backmatter not in self.sections:
         raise error.NoSuchSection(backmatter)
+
     # Use explicit order as partial ordering and merge with default section
     # ordering. All custom sections must be ordered explicitly.
     self.order = self._GetSectionOrder(self.order, self.sections)
 
+    # Child section collection
+    to_delete = []
+    for key in self.sections:
+      section = self.sections[key]
+      if 'parent_id' in section.locals and section.locals['parent_id']:
```

Comment:
I'd avoid the `try` / `except KeyError` over the whole block when it's just one of several lookups you're trying to catch (a typo in 'children' or something could result a misleading `NoSuchSection(parent_id)`. Also you can use setdefault and other python idioms to tighten up the code so it's easier to understand at a glance (to my eyes, anyway):
```python
parent_id = section.locals.get('parent_id', None)
if parent_id:
  if parent_id not in self.sections:
    raise error.NoSuchSection(parent_id)
  parent = self.sections[parent_id]
  parent.locals.setdefault('children', []).append(section)
  to_delete.append(key)
```

(That's one suggested alternative. Feel free to pick and chooses parts of that to suit your taste.)